test/forms.e2e-spec.ts

```diff
@@ -0,0 +1,95 @@
+import { INestApplication, ValidationPipe } from "@nestjs/common";
+import { Test, TestingModule } from "@nestjs/testing";
+import { AppModule } from "../src/app.module";
+import { seed } from "../prisma/seed/seed";
+import * as request from "supertest";
+import * as cookieParser from "cookie-parser";
+import { extractCookieByKey } from "./utils";
+
+const loginUrl = "/auth/login";
+
+const loginAndGetTokens = async (
+    email: string,
+    password: string,
+    app: INestApplication,
+) => {
+    const r = await request(app.getHttpServer()).post(loginUrl).send({
+        email,
+        password,
+    });
+
+    const access_token = extractCookieByKey(
+        r.headers["set-cookie"],
+        "access_token",
+    );
+    const refresh_token = extractCookieByKey(
+        r.headers["set-cookie"],
+        "refresh_token",
+    );
+
+    return { access_token, refresh_token };
+};
+
+describe("FormController e2e Tests", () => {
+    let app: INestApplication;
+
+    beforeAll(async () => {
+        const moduleFixture: TestingModule = await Test.createTestingModule({
+            imports: [AppModule],
+        }).compile();
+
+        await seed();
+
+        app = moduleFixture.createNestApplication();
+        app.useGlobalPipes(new ValidationPipe());
+        app.use(cookieParser());
+        await app.init();
+    });
+
+    afterAll(async () => {
+        await app.close();
+    });
+
+    describe("Get Forms Endpoint Testing", () => {
```

Comment:
could you have one "describe" for each endpoint? You can look at auth or techs e2e tests for example

Reply:
sure no problem!

Reply:
I have added one more 'describe' block for handling GET requests to retrieve a specific form (GET /forms/:formId). So, in the test suite, we now have one for handling GET requests to retrieve all forms (GET ALL /forms) and another for handling GET requests for a specific form (GET /forms/:formId), including a case to handle invalid form IDs. I don't see the need to have separate blocks for each endpoint, as /forms/${invalidFormId} is also part of retrieving a specific form. Let me know what you think :)

Reply:
yes, for your case, there's no need to have nested describe